File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/GetRoundsUseCase.java
package es.etologic.mahjongscoring2.domain.use_cases;

import java.util.List;

import javax.inject.Inject;

import es.etologic.mahjongscoring2.data.repositories.RoundsRepository;
import es.etologic.mahjongscoring2.domain.model.Round;
import io.reactivex.Single;

public class GetRoundsUseCase {

    private final RoundsRepository roundsRepository;

    @Inject
    public GetRoundsUseCase(RoundsRepository roundsRepository) {
        this.roundsRepository = roundsRepository;
    }

    public Single<List<Round>> getGameRounds(long gameId) {
        return roundsRepository.getRoundsByGame(gameId);
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/GetRoundsUseCase.kt
package es.etologic.mahjongscoring2.domain.use_cases

import es.etologic.mahjongscoring2.data.repositories.RoundsRepository
import es.etologic.mahjongscoring2.domain.model.Round
import io.reactivex.Single
import javax.inject.Inject

class GetRoundsUseCase @Inject
constructor(private val roundsRepository: RoundsRepository) {
    
    fun getGameRounds(gameId: Long): Single<List<Round>> {
        return roundsRepository.getRoundsByGame(gameId)
    }
}
